app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -233,10 +233,22 @@ public void onResume()
         // '1' is the brightest. We attempt to maximize the brightness
         // to help barcode readers scan the barcode.
         Window window = getWindow();
-        if(window != null && settings.useMaxBrightnessDisplayingBarcode())
+        if(window != null)
         {
             WindowManager.LayoutParams attributes = window.getAttributes();
-            attributes.screenBrightness = 1F;
+
+            if (settings.useMaxBrightnessDisplayingBarcode())
+            {
+                attributes.screenBrightness = 1F;
+            }
+
+            if (settings.getKeepScreenOn())
+            {
+                window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON|
+                        WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD|
+                        WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED);
```

Comment:
These are kinda two different features:

1. Keep screen on
2. Show over lock screen

What do you think about turning this into 2 different options?

With "Keep screen on" only setting FLAG_KEEP_SCREEN_ON and show over lock screen activating FLAG_DISMISS_KEYGUARD and FLAG_SHOW_WHEN_LOCKED?